modules/local/matricescreation.nf

```diff
@@ -0,0 +1,48 @@
+process MATRICESCREATION {
+    label 'process_single'
+
+    conda 'r-ggplot2=3.4.3 bioconductor-shortread=1.58.0 r-ggpubr=0.6.0 r-ggmsa=1.0.2 r-seqmagick=0.1.6 r-tidyr=1.3.0 r-ggseqlogo=0.1 r-cowplot=1.1.1 r-seqinr=4.2_30 r-optparse=1.7.3 r-dplyr=1.1.2 r-plyr=1.8.8 r-stringr=1.5.0 r-plotly=4.10.2'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-6de07928379e6eface08a0019c4a1d6b5192e805:0d77388f37ddd923a087f7792e30e83ab54c918c-0' :
+        'biocontainers/mulled-v2-6de07928379e6eface08a0019c4a1d6b5192e805:0d77388f37ddd923a087f7792e30e83ab54c918c-0' }"
+
+    input:
+    val(meta)
+
+    output:
+    tuple val(meta), path("*.txt"), emit: design_matrix
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    meta.id = "${meta.treatment}_vs_${meta.reference}"
+
+    """
+    #!/usr/bin/env Rscript
```

Comment:
hey @itrujnara i'm looking into this now, and i was wondering if it's not a bit less readable since we have variables? Would you just take the whole R script as it is right now and put it in `matricecreations.R` in bin for instance?

Reply:
I would say making it a template under `templates` is the best in this case. This way, you can continue using the same nextflow variable interpolation.

Reply:
i just find it's less readable to understand what the module is doing? Is there a specific reason why we don't want any code in the script part?